Add prometheus metrics to monitor replication

JohnStrunk opened the pull request:
**Describe what this PR does**
This adds metrics to enable monitoring the state of replication relationships via prometheus

**Is there anything that requires special attention?**
<!-- Do you have any questions? Did you do something clever? -->

**Related issues:**
Fixes #108

JohnStrunk commented:
cc: @mulbc

JohnStrunk commented:
Metrics example:

![Screenshot from 2021-03-12 13-38-46](https://user-images.githubusercontent.com/6456608/110984459-290ca080-8339-11eb-8ccd-da50808b4563.png)

JohnStrunk commented:
Metrics example:

![Screenshot from 2021-03-12 13-38-46](https://user-images.githubusercontent.com/6456608/110984459-290ca080-8339-11eb-8ccd-da50808b4563.png)

JohnStrunk commented:
Metrics example:

![Screenshot from 2021-03-12 13-38-46](https://user-images.githubusercontent.com/6456608/110984459-290ca080-8339-11eb-8ccd-da50808b4563.png)

mulbc commented on docs/usage/metrics/index.rst:
```diff
@@ -0,0 +1,240 @@
+   example, when using the rsync mover with a schedule on the source but not on
+   the destination, only the metric for the source side is meaningful.
+scribe_sync_duration_seconds
+   This is a summary of the time required for each sync iteration. My monitoring
```
`By monitoring...`

mulbc commented:
Could you add an example scrape of your exporter here or (even better) in the documentation? That way it's easier for people knowing Prometheus to know what they should expect...
Also I'll ping my friends in the monitoring ENG to review the monitoring end point to check if they are sane (it's a DB after all and optimizations should happen as early as possible)

JohnStrunk commented:
I added some raw metric output to the docs. Copied here:
```
$ curl -s http://127.0.0.1:8080/metrics | grep scribe
# HELP scribe_missed_intervals_total The number of times a synchronization failed to complete before the next scheduled start
# TYPE scribe_missed_intervals_total counter
scribe_missed_intervals_total{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 0
scribe_missed_intervals_total{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 0
# HELP scribe_sync_duration_seconds Duration of the synchronization interval in seconds
# TYPE scribe_sync_duration_seconds summary
scribe_sync_duration_seconds{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination",quantile="0.5"} 180.722389767
scribe_sync_duration_seconds{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination",quantile="0.9"} 544.86628289
scribe_sync_duration_seconds{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination",quantile="0.99"} 544.86628289
scribe_sync_duration_seconds_sum{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 1905.815430317
scribe_sync_duration_seconds_count{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 9
scribe_sync_duration_seconds{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source",quantile="0.5"} 9.756509957
scribe_sync_duration_seconds{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source",quantile="0.9"} 19.624571877
scribe_sync_duration_seconds{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source",quantile="0.99"} 19.624571877
scribe_sync_duration_seconds_sum{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 96.962036
scribe_sync_duration_seconds_count{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 9
# HELP scribe_volume_out_of_sync Set to 1 if the volume is not properly synchronized
# TYPE scribe_volume_out_of_sync gauge
scribe_volume_out_of_sync{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 0
scribe_volume_out_of_sync{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 0
```

What I'm not sure about is what I've chosen to do w/ the `obj_` prefix. When scraped by prometheus, at least in ocp, a namespace label gets automatically added w/ the namespace where the metric was acquired (scribe-system). This overwrites the `namespace` label I had been using to designate where the CR was located--- So I've added the `obj_` prefix to both `name` and `namespace` as they refer to the object that the metric is about.
